### DISCUSSION TITLE
(RISOLTO) Integrare-telecamere-come-sensori-di-movimento

### DISCUSSION THREAD

**Discussion by calz64** [indomus/forum#464] [HUB: Home Assistant]
Buongiorno ho seguito la guida https://indomus.it/guide/integrare-telecamere-come-sensori-di-movimento-su-home-assistant-via-motioneye/  per creare/usare le camere come sensori di movimento, ha funzionato tutto alla perfezione ma poi oggi mi trovo che i due binary_sensor della configurazione sotto non risultano più disponibile e non rilevano più 'on' e 'off'.
 (se ascolto con mosquitto_sub vedo invece le pubblicazioni sul broker)
```
binary_sensor:
  - platform: mqtt
    name: Movimento Sala
    payload_on: "on"
    payload_off: "off"
    device_class: motion
    state_topic: "stat/motionEye/motion"
    value_template: >
      {% if value_json.ipcam == 'sala_on' %}
        on
      {% elif value_json.ipcam == 'sala_off' %}
        off
      {% else %}
        {{ states('binary_sensor.movimento_sala') }}
      {% endif %}

  - platform: mqtt
    name: Movimento Loggia
    payload_on: "on"
    payload_off: "off"
    device_class: motion
    state_topic: "stat/motionEye/motion"
    value_template: >
      {% if value_json.ipcam == 'loggia_on' %}
        on
      {% elif value_json.ipcam == 'loggia_off' %}
        off
      {% else %}
        {{ states('binary_sensor.movimento_loggia') }}
      {% endif %}
```
![Cattura](https://user-images.githubusercontent.com/84615011/123206883-c345be00-d4bc-11eb-82fb-bd7db891ddb4.JPG)

**Comment by calz64** [comment]
Forse il problema non è nelle entità binary-sensor ma forse sta in questo LOG???
`2021-06-24 21:12:30 ERROR (Thread-3) [homeassistant.components.mqtt] Unable to connect to the MQTT broker: Connection Refused: not authorised.`

Ho letto googolando che spesso è un problema di password? qualcuno sa darmi una dritta.

**Reply by snakuzzo** [reply]
non solo...potrebbe anche dipendere dalla lunghezza del topic o del client id.
Posta configurazione del broker MQTT e del client che si collega

**Reply by calz64** [reply]
spero di aver capito cosa mi stai chiedendo:

```
logins:
  - username: mqtt_user
    password: XXXXXX
customize:
  active: false
  folder: mosquitto
certfile: fullchain.pem
keyfile: privkey.pem
require_certificate: false
anonymous: false
```

il client è motionEye quindi il topic è 
`stat/motionEye/motion -m '{"ipcam":"sala_on"}'`

**Reply by KC-inDomus** [reply]
eh ma nel comando che hai impostato su motioneye non gli passi le credenziali per accedere al broker:
come pensi possa autenticarsi?

**Reply by calz64** [reply]
No quello che ho scritto è solo il topic (ma forse non è quello che mi chiede snakuzzo che parla si di lunghezza topic ma poi chiede config. del client), in motionEye ho attivato le finestre "Run a commnad" e inserito (come da GUIDA)
`mosquitto_pub -h 192.168.178.35 -u mqtt_user -P '.....pw....' -t stat/motionEye/motion -m '{"ipcam":"sala_on"}'`  (e poi anche l'omologo "sala_off)
Il fatto è che se nel terminale mi metto in ascolto con il comando mosquitto_sub vedo il client motionEye che pubblica sul broker, ma non riesco a farlo leggere dal sensore binary_sensor che rimane su 'unavailable'. E' per questo che ho pensato che il problema non è del broker ma del messaggio LOG che dice che HA è "unable to connet to the MQTT".
E' veramente frustrante, anche perchè all'inizio ha funzionato poi puff...
![Cattura](https://user-images.githubusercontent.com/84615011/123501879-5101e480-d648-11eb-9b63-fd464febf09a.JPG)

**Reply by calz64** [reply]
Altro dubbio: nella guida MQTT dice che se il broker è configurato bene in "strumenti per sviluppatori" appare la finestra "MQTT" io non vedo nulla significa qualcosa?
![Cattura](https://user-images.githubusercontent.com/84615011/123502092-dfc33100-d649-11eb-945a-78065425dcba.JPG)

**Reply by snakuzzo** [reply]
Da quello che vedo e leggo non hai problemi di connessione al broker. Hai problemi a creare l'entità su HA. Devi capire bene come farlo.
Quanto alla console per MQTT, se lo fai da riga di comando è sicuramente meglio che farlo da gui. A tal proposito dai una letta [QUI](https://indomus.it/formazione/mosquitto-mqtt-broker-comandi-utili/)

Ad ogni modo quello che cerchi lo trovi sotto CONFIGURATION > INTEGRATIONS > MQTT

**Reply by calz64** [reply]
> ---  _**Da quello che vedo e leggo non hai problemi di connessione al broker. Hai problemi a creare l'entità su HA. Devi capire bene come farlo.**_
Da quello che ho capito , e forse è poca roba, l'entità binary_sensor si è creata aggiungendo le righe in configuration.yaml (e che io ho preso dalla guida, da solo non sarei stato capace.)
```
binary_sensor:
  - platform: mqtt
    name: Movimento Sala
    payload_on: "on"
    payload_off: "off"
    device_class: motion
    state_topic: "stat/motionEye/motion"
    value_template: >
      {% if value_json.ipcam == 'sala_on' %}
        on
      {% elif value_json.ipcam == 'sala_off' %}
        off
      {% else %}
        {{ states('binary_sensor.movimento_sala') }}
      {% endif %}
..... e poi quella con loggia (come sopra riporatato per intero)
```
> ---- _**Quanto alla console per MQTT, se lo fai da riga di comando è sicuramente meglio che farlo da gui. A tal proposito dai una letta [QUI](https://indomus.it/formazione/mosquitto-mqtt-broker-comandi-utili/)**_
> 
la guida l'ho letta e riletta e i comandi scritti nel terminale sono quelli che ho imparato dalla guida, e in effetti ho capito che il broker funziona, il clinet motionEye anche perchè pubblica. Quello che non va, o sembra non andare, è il client (se cosi si può dire/chiamare) binary_sensor che non recepisce le pubblicazioni sul broker e pensavo fosse un problema del log
`ERROR (Thread-3) [homeassistant.components.mqtt] Unable to connect to the MQTT broker: Connection Refused: not authorised.`
Cioè che fosse inibito a questo binary_sensor di leggere i messaggi sul broker, ma se invece è un problema di entità e di configurazione (quella scritta sopra e presa dalla guida) avrei bisogno di qualcuno che capisse bene il codice..

> --- _**Ad ogni modo quello che cerchi lo trovi sotto CONFIGURATION > INTEGRATIONS > MQTT**_
Scusa ma non ho capito cosa indica questo " configuration>integrations>mqtt " è un percorso, dove lo posso trovare.
E' una cosa che ho letto dalla guida che mi hai indicato ma forse mal interpretato.
![Cattura](https://user-images.githubusercontent.com/84615011/123519690-2e55e700-d6ad-11eb-9dcb-956f5cfa560d.JPG)


In ogni caso GRAZIE della pazienza.

**Reply by calz64** [reply]
Eureka.... come al solito una banale disattenzione, convinto che fosse la password come suggerito all'inizio del post, ho cambiato password in configurazione di "mosquitto broker" di supervisor, ma non avevo ri-configurato l'integrazione di MQTT che non segnalava nulla nella "card" dell'integrazione ma che nel LOG a questo punto risulta chiaro. WOW
